dimod/generators/random.py

```diff
@@ -500,3 +502,63 @@ def doped(p: float, graph: GraphLike,
         bqm.add_interaction(u, v, J)
 
     return bqm
+
+
+
+@graph_argument('graph')
+def power_r(r: int, graph: GraphLike,
+          seed: Union[None, int, np.random.Generator] = None,) -> BinaryQuadraticModel:
```

Comment:
Careful with the indentation. We should also make `seed` a keyword only argument.

```suggestion
def power_r(r: int, 
            graph: GraphLike,
            *,
            seed: Union[None, int, np.random.Generator] = None,
            ) -> BinaryQuadraticModel:
```

Reply:
is it ok to use the formatting that Black suggests instead?

```python
@graph_argument("graph")
def power_r(
    r: int,
    graph: GraphLike,
    *,
    seed: Union[None, int, np.random.Generator] = None,
) -> BinaryQuadraticModel:


```

Reply:
Hah, technically a pep8 violation (https://github.com/psf/black/issues/1178 for the rabbit hole), but sure, that's OK.